amxmodx/commandline.cpp

```diff
@@ -0,0 +1,100 @@
+#include "amxmodx.h"
+
+#define CHECK_COMMANDLINE_AVAILABILITY()                                        \
+    if (!IsCommandLineAvailable())                                              \
+    {                                                                           \
+        LogError(amx, AMX_ERR_NATIVE, "The command line is not available.");    \
+        return 0;                                                               \
+    }
+
+static bool IsCommandLineAvailable()
+{
+    return g_engfuncs.pfnCheckParm != nullptr;
```

Comment:
This check won't work, because amxmodx will just crash in `GiveFnptrsToDll` or `pfnCheckParm` will be filled with garbage. I mean old metamods, like 1.19 where `pfnCheckParm` isn't present.

Reply:
What solution do you propose?

Reply:
@rsKliPPy looks like metamod developers didn't update any API version that can be checked in plugin, so there is no legal way.

Reply:
So what do we do? Scrap this? Bump Metamod requirement? Something else?

Reply:
In 1.8.2 release notes, it was clearly stated to upgrade to metamod 1.21-am.
This would not be a surprise that 1.9 would require the same version.

Documenting this properly would be likely enough, but having a way to gracefully warn the user would be welcomed though. If both metamod could bump the API version, we could make 1.9 requiring this version.

Reply:
We can remove `IsCommandLineAvailable` native function if amxx supports only last hlds build and metamod >= 1.21-am.